### PR TITLE
Java now uses struct_tag types (2nd edition)

### PR DESCRIPTION
**Pull request by allredj**
Replacement for https://github.com/diffblue/cbmc/pull/3102 with the following changes:
* Added a missing condition in `interpretert::get_size()` (last commit)
* Rebased on cbmc/develop

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by allredj** [comment]
TG bump is here: https://github.com/diffblue/test-gen/pull/2384

File: jbmc/src/java_bytecode/java_types.cpp [hunk 1]
```diff
@@ -164,7 +164,7 @@ bool is_multidim_java_array_type(const typet &type)
 {
   return is_java_array_type(type) &&
          is_java_array_type(
-           java_array_element_type(to_symbol_type(type.subtype())));
```
Comment (peterschrammel) [inline jbmc/src/java_bytecode/java_types.cpp:167]: clang-format nitpick: https://travis-ci.org/diffblue/cbmc/jobs/463928375#L538

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 0b2d3b164b6d7250b6f0d24c512ca2011d389767).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/93796858
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by kroening — APPROVED** [review]

File: jbmc/src/java_bytecode/java_types.cpp (hunk 1 above)
Comment (allredj) [reply jbmc/src/java_bytecode/java_types.cpp:167]: @peterschrammel fixed

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**